subsys/testsuite/ztest/Kconfig

```
@@ -150,22 +150,41 @@ config ZTEST_SHUFFLE
 
 if ZTEST_SHUFFLE
 config ZTEST_SHUFFLE_SUITE_REPEAT_COUNT
-	int "Number of iterations the test suite will run"
+	int "[DEPRECATED] Number of iterations the test suite will run"
 	default 3
```

```suggestion
	select DEPRECATED
	default 3
```
and same for `ZTEST_SHUFFLE_TEST_REPEAT_COUNT`

---

> and same for `ZTEST_SHUFFLE_TEST_REPEAT_COUNT`

After adding the select field DEPRECATED it throws the error while building.

```
warning: the int symbol ZTEST_SHUFFLE_SUITE_REPEAT_COUNT (defined at subsys/testsuite/ztest/Kconfig:152) has selects or implies

warning: the int symbol ZTEST_SHUFFLE_TEST_REPEAT_COUNT (defined at subsys/testsuite/ztest/Kconfig:160) has selects or implies

warning: The int symbol ZTEST_SHUFFLE_SUITE_REPEAT_COUNT (defined at subsys/testsuite/ztest/Kconfig:152) is being evaluated in a logical context somewhere. It will always evaluate to n.

warning: The int symbol ZTEST_SHUFFLE_TEST_REPEAT_COUNT (defined at subsys/testsuite/ztest/Kconfig:160) is being evaluated in a logical context somewhere. It will always evaluate to n.
```

@tomi-font Do you have an ideas, how to fix it?

---

this warning might be because these symbols with `select DEPRECATED` are of int not bool type.

---

@gbarkadiusz You did not manage to get rid of the warnings? Do they prevent a successful build?

---

Sadly not... It seems that the select can be used only with the bool typed config?

---

BTW, It's nice to have any guide regarding the process of deprecating Kconfig ;)

---

Yeah, it's true. I couldn't find anything from the docs with a quick search. @kartben ?